sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -239,25 +239,26 @@ public ConfigurationClientBuilder endpoint(String endpoint) {
      * @param connectionString Connection string in the format "endpoint={endpoint_value};id={id_value};
      * secret={secret_value}"
      * @return The updated ConfigurationClientBuilder object.
-     * @throws NullPointerException If {@code credential} is {@code null}.
+     * @throws IllegalArgumentException if {@code connectionString} is null or an empty string. Or the secret is
+     * invalid and cannot instantiate HMAC-SHA256 MAC algorithm.
      */
     public ConfigurationClientBuilder connectionString(String connectionString) {
-        Objects.requireNonNull(connectionString);
+        if (CoreUtils.isNullOrEmpty(connectionString)) {
```

Comment:
According to the design spec, null should throw NullPointerException and isEmpty should throw IllegalArgumentException. I'd split this into two steps.

https://azure.github.io/azure-sdk/java_implementation.html#java-errors-system-errors